geode-core/src/main/java/org/apache/geode/cache/client/internal/QueueManagerImpl.java

```diff
@@ -1030,8 +1030,10 @@ private boolean addToConnectionList(QueueConnectionImpl connection, boolean isPr
       try {
         connection.internalClose(pool.getKeepAlive());
       } catch (Exception e) {
-        logger.warn("Error destroying client to server connection to {}", connection.getEndpoint(),
```

Comment:
Was the warning not needed anymore?

Reply:
If internalClose is called after internalDestroy for the same QueueConnectionImpl, it used to throw a NullPointerException that was logged by that warn message. After changing the code to check for null, now the internalClose method throws a ConnectionDestroyedException which would be logged by that same warn message. Looking at that message, I don't think it needs to be logged at a warn level since the QueueManagerImpl addToConnectionList method only gets into that code path when the QueueManagerImpl or the Cache is shutting down.

btw - this is a great interview question. If a try block throws an exception and its finally block also throws an exception, which one is seen by the caller? The finally block exception hides the try block exception.

Reply:
Interesting try/catch factoid